### ISSUE TITLE
Cuda build fails on Ubuntu 19.04 (with Cuda 10.x)

### ISSUE THREAD

**Issue by declension** [AccelerateHS/accelerate-llvm#51]
I am submitting a...
- [x] bug report
- [ ] feature request
- [ ] support request => you might also like to ask your question on the [mailing list](mailto:accelerate-haskell@googlegroups.com) or [gitter chat](https://gitter.im/AccelerateHS/Lobby).

## Description
Build for `accelerate-llvm-ptx` fails on Ubuntu 19 with latest Cuda

## Expected behaviour
Working build.

## Current behaviour
Fails with this:
```
accelerate-llvm-ptx> /tmp/stack21219/accelerate-llvm-ptx-1.2.0.1/src/Data/Array/Accelerate/LLVM/PTX/Compile/Libdevice/Load.hs:122:23: error:
accelerate-llvm-ptx>     • Exception when trying to run compile-time code:
accelerate-llvm-ptx>         /usr/nvvm/libdevice: getDirectoryContents:openDirStream: does not exist (No such file or directory)
accelerate-llvm-ptx>       Code: libdeviceBitcode (Compute 3 5)
accelerate-llvm-ptx>     • In the Template Haskell splice $$(libdeviceBitcode (Compute 3 5))
accelerate-llvm-ptx>       In the expression: $$(libdeviceBitcode (Compute 3 5))
accelerate-llvm-ptx>       In an equation for ‘libdevice_35_bc’:
accelerate-llvm-ptx>           libdevice_35_bc = $$(libdeviceBitcode (Compute 3 5))
accelerate-llvm-ptx>     |
accelerate-llvm-ptx> 122 | libdevice_35_bc = $$( libdeviceBitcode (Compute 3 5) )
accelerate-llvm-ptx>     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
accelerate-llvm-ptx>
```

## Possible solution (optional)
Some workarounds mentioned on [this mailing list thread from 2017](https://groups.google.com/forum/#!topic/accelerate-haskell/DGbZvxmKlOs) but no real progress. 

Seems to be a matter of convincing TH where to find `libdevice` (or at least that's the _first_ step) given a different directory / file structure.

## Steps to reproduce (for bugs)
 1. Install latest Cuda (currently 10.1) on latest Ubuntu (19.04) via `sudo apt install nvidia-cuda-toolkit`. Aside: I started trying to downgrade everything, but this itself brought more problems, and finally having `apt` support with all-new tooling is an attractive thing to keep... 
 2. Set up a minimal accelerate project, e.g.:
```Stack
resolver: lts-14.3
packages:
- .
extra-deps:
- accelerate-1.2.0.1
- accelerate-examples-1.2.0.1
- accelerate-llvm-1.2.0.1
- accelerate-llvm-native-1.2.0.1
- accelerate-llvm-ptx-1.2.0.1
```
 3. `stack build`

## Your environment
Ubuntu 19.04 (though think this would happen on 18.x too)
Cuda / `nvcc` version `10.1.105`

**Comment by tmcdonell** [comment]
On my 19.10 box this is located at `/usr/lib/nvidia-cuda-toolkit/libdevice/libdevice.10.bc`

**Comment by sakehl** [comment]
I suspect that it is this line:
https://github.com/AccelerateHS/accelerate-llvm/blob/0f92cd6ad324f87378bffbb9b74bbd0722cee417/accelerate-llvm-ptx/src/Data/Array/Accelerate/LLVM/PTX/Compile/Libdevice/TH.hs#L121
I just came across something similar. (Althouh on another branch).

It looks like cudaInstallPath is returning /usr/ instead of /usr/lib/cuda or whatever the correct path should be.

**Comment by tmcdonell** [comment]
I'll fix this in the `nvvm` library
/ping AccelerateHS/accelerate#444

**Comment by tmcdonell** [comment]
Fixed in the new 1.3 release
